ts/components/conversation/ConversationRequestButtons.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import styled from 'styled-components';
+import { getMessageCountByType } from '../../data/data';
+import {
+  approveConvoAndSendResponse,
+  declineConversationWithConfirm,
+} from '../../interactions/conversationInteractions';
+import { MessageDirection } from '../../models/messageType';
+import { getConversationController } from '../../session/conversations';
+import { getSelectedConversation } from '../../state/selectors/conversations';
+import { SessionButton, SessionButtonColor, SessionButtonType } from '../basic/SessionButton';
+
+export const ConversationMessageRequestButtons = () => {
+  const selectedConversation = useSelector(getSelectedConversation);
+  const [hasIncoming, setHasIncomingMsgs] = useState(false);
+  const [incomingChecked, setIncomingChecked] = useState(false);
+
+  useEffect(() => {
+    async function getIncomingMessages() {
+      const id = selectedConversation?.id;
+      if (id) {
+        const msgCount = await getMessageCountByType(
+          selectedConversation?.id,
+          MessageDirection.incoming
+        );
+        if (msgCount > 0) {
+          setHasIncomingMsgs(true);
+        } else {
+          setHasIncomingMsgs(false);
+        }
+        setIncomingChecked(true);
+      }
+    }
+    // tslint:disable-next-line: no-floating-promises
+    getIncomingMessages();
+  });
+
```

Comment:
because there is no `[]` as dependency of useeffect, this code is run on every single render, this is not ideal. Why do you need this?